tests/test_balances.py

```diff
@@ -0,0 +1,170 @@
+
+    balance = client.balances.get(BALANCE_ID)
+    balance_report = balance.report.get(primary=True)
+    assert isinstance(balance_report, BalanceReport)
+
+    assert balance_report.resource == "balance-report"
+    assert balance_report.balance_id == BALANCE_ID
+    assert balance_report.time_zone == "Europe/Amsterdam"
+    assert balance_report.date_from == "2021-01-01"
+    assert balance_report.date_until == "2021-01-31"
+    assert balance_report.grouping == "transaction-categories"
+    assert balance_report.totals == {"some_data": "some_data"}
+
+
+def test_get_balance_transaction(client, response):
+    """Get a balance report."""
+    response.get(f"https://api.mollie.com/v2/balances/{BALANCE_ID}", "balance_single")
+    response.get(f"https://api.mollie.com/v2/balances/{BALANCE_ID}/transactions", "balance_transactions_list")
+
+    balance = client.balances.get(BALANCE_ID)
+    balance_transactions = balance.transactions.list()
+    assert_list_object(balance_transactions, BalanceTransaction)
+
+    balance_transaction: BalanceTransaction = balance_transactions[0]
+    assert balance_transaction.resource == "balance_transaction"
+    assert balance_transaction.id == BALANCE_TRANSACTION_ID
+    assert balance_transaction.type == "refund"
+    assert balance_transaction.result_amount == {"value": "-10.25", "currency": "EUR"}
+    assert balance_transaction.initial_amount == {"value": "-10.00", "currency": "EUR"}
+    assert balance_transaction.deductions == {"value": "-0.25", "currency": "EUR"}
+
+    assert balance_transaction.created_at == "2021-01-10T12:06:28+00:00"
+    assert balance_transaction.context == {"paymentId": "tr_7UhSN1zuXS", "refundId": "re_4qqhO89gsT"}
+
+
+def test_get_primary_balance_transaction(client, response):
+    """Get the primary balance report."""
+    response.get(f"https://api.mollie.com/v2/balances/{BALANCE_ID}", "balance_single")
+    response.get(f"https://api.mollie.com/v2/balances/{BALANCE_ID}/primary/transactions", "balance_transactions_list")
+
+    balance = client.balances.get(BALANCE_ID)
+    balance_transactions = balance.transactions.list(primary=True)
+    assert_list_object(balance_transactions, BalanceTransaction)
+
+    balance_transaction: BalanceTransaction = balance_transactions[0]
+    assert balance_transaction.resource == "balance_transaction"
+    assert balance_transaction.id == BALANCE_TRANSACTION_ID
+    assert balance_transaction.type == "refund"
+    assert balance_transaction.result_amount == {"value": "-10.25", "currency": "EUR"}
+    assert balance_transaction.initial_amount == {"value": "-10.00", "currency": "EUR"}
+    assert balance_transaction.deductions == {"value": "-0.25", "currency": "EUR"}
+
+    assert balance_transaction.created_at == "2021-01-10T12:06:28+00:00"
+    assert balance_transaction.context == {"paymentId": "tr_7UhSN1zuXS", "refundId": "re_4qqhO89gsT"}
+
+
+def test_balances_base_validate_resource_id(client):
```

Comment:
You are not testing that we are validating the Balance id in all the correct place. Please see the tests that end in `..._invalid_id` in f.i. `test_orders.py`